File: src/lib/bindings/tracks.ts
import { invoke } from "@tauri-apps/api/core";

export interface TrackInfo {
  id: number;
  track_type: string;
  title: string;
  lang: string;
  selected: boolean;
}

export const getTracks = () => invoke<TrackInfo[]>("get_tracks");
export const selectSubtitle = (id: number) => invoke<void>("select_subtitle", { id });
export const selectAudioTrack = (id: number) => invoke<void>("select_audio_track", { id });
export const loadSubtitle = (path: string) => invoke<void>("load_subtitle", { path });
export const toggleSubtitles = () => invoke<void>("toggle_subtitles");
export const setSubtitleDelay = (seconds: number) => invoke<void>("set_subtitle_delay", { seconds });
export const setAudioDelay = (seconds: number) => invoke<void>("set_audio_delay", { seconds });

File: src-tauri/src/commands/tracks.rs
use tauri::State;

use crate::error::AppError;
use crate::services::tracks::{TrackInfo, TracksService};
use crate::state::MpvState;

#[tauri::command]
pub async fn get_tracks(state: State<'_, MpvState>) -> Result<Vec<TrackInfo>, AppError> {
    let mpv = state.get()?;
    Ok(TracksService::get_all(mpv))
}

#[tauri::command]
pub async fn select_subtitle(id: i64, state: State<'_, MpvState>) -> Result<(), AppError> {
    TracksService::select_subtitle(state.get()?, id)?;
    Ok(())
}

#[tauri::command]
pub async fn select_audio_track(id: i64, state: State<'_, MpvState>) -> Result<(), AppError> {
    TracksService::select_audio(state.get()?, id)?;
    Ok(())
}

#[tauri::command]
pub async fn load_subtitle(path: String, state: State<'_, MpvState>) -> Result<(), AppError> {
    TracksService::load_subtitle(state.get()?, &path)?;
    Ok(())
}

#[tauri::command]
pub async fn toggle_subtitles(state: State<'_, MpvState>) -> Result<(), AppError> {
    TracksService::toggle_subtitles(state.get()?)?;
    Ok(())
}

#[tauri::command]
pub async fn set_subtitle_delay(seconds: f64, state: State<'_, MpvState>) -> Result<(), AppError> {
    TracksService::set_subtitle_delay(state.get()?, seconds)?;
    Ok(())
}

#[tauri::command]
pub async fn set_audio_delay(seconds: f64, state: State<'_, MpvState>) -> Result<(), AppError> {
    TracksService::set_audio_delay(state.get()?, seconds)?;
    Ok(())
}
